### PR TITLE
Add multi-platform support with buildx

### PR DESCRIPTION
**Pull request by ericwill**
* Cleaned up the build action to use a loop
* Added support for multi-platform builds via docker buildx

Signed-off-by: Eric Williams <ericwill@redhat.com>

### CODE REVIEW

File: .github/workflows/build-publish.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+      - name: Docker Buildx
+        uses: crazy-max/ghaction-docker-buildx@v1.4.0
+        with:
+          version: v0.3.1
```
Comment (benoitf) [inline .github/workflows/build-publish.yaml]: we can bump to v0.4.1 now (I tried a couple of days ago)

File: .github/workflows/build-publish.yaml [hunk 2]
```diff
@@ -19,13 +23,13 @@ jobs:
...
-          docker push "${IMAGE_NAME}:${IMAGE_VERSION}"
-          docker tag "${IMAGE_NAME}" "${IMAGE_NAME}:${IMAGE_VERSION}-${SHORT_SHA1}"
-          echo "Pushing ${IMAGE_NAME}:${IMAGE_VERSION}-${SHORT_SHA1}"
-          docker push "${IMAGE_NAME}:${IMAGE_VERSION}-${SHORT_SHA1}"
+          docker buildx build --platform ${PLATFORMS} --cache-from=type=registry,ref=${IMAGE_NAME}--cache-to=type=registry,ref=${IMAGE_NAME},mode=max -t ${IMAGE_NAME} .
+          declare -a tags=(${IMAGE_NAME}:${IMAGE_VERSION} ${IMAGE_NAME}:${IMAGE_VERSION}-${SHORT_SHA1})
+          for tag in ${tags[@]}; do
+            echo "Tagging and pushing $tag"
+            docker tag "${IMAGE_NAME}" "$tag"
+            docker push "$tag"
```
Comment (benoitf) [inline .github/workflows/build-publish.yaml]: I think it doesn't work like that, image should be pushed by docker buildx with --push 
And for aliases, need to use docker manifest create or something else
docker tag can only work with a local image which is not multi-arch

Comment (ericwill) [reply .github/workflows/build-publish.yaml]: Given that we will be switching to digests everywhere, do we even care about aliases? Why not just push it as `IMAGE_VERSION-SHORT_SHA1`?

Comment (benoitf) [reply .github/workflows/build-publish.yaml]: yes I'm fine

**Review by benoitf — APPROVED** [review]
approving for upcoming changes as it's late in France :-)
And it's a day off tomorrow in France 🇫🇷

**Comment by ericwill** [comment]
> approving for upcoming changes as it's late in France :-)
> And it's a day off tomorrow in France fr

Enjoy :)

**Comment by benoitf** [comment]
we might also require a job that doesn't push && do the login to check PR (see if it compiles or not)

**Comment by ericwill** [comment]
> we might also require a job that doesn't push && do the login to check PR (see if it compiles or not)

Right, that would be in `pr-check.yaml`? I'll do the conversion to buildx as well.

**Comment by benoitf** [comment]
yes, another .yaml or add conditional in jobs
like (if not against master branch)

**Comment by benoitf** [comment]
https://github.community/t5/GitHub-Actions/run-step-on-if-branch-tag-is/m-p/38897#M3566

and use `$PUSH`arg to buildx with '--push' if master else ''

**Comment by benoitf** [comment]
pros/cons of single yaml with conditions vs two yaml with a lot of copy/paste

**Comment by ericwill** [comment]
@benoitf if no objections please merge, I'm going to start converting the other repos next week.

**Review by benoitf — APPROVED** [review]
You can merge. Let's see

**Pull request merged**